restapi/src/main/java/io/stargate/web/docsapi/dao/Paginator.java

```diff
@@ -0,0 +1,106 @@
+package io.stargate.web.docsapi.dao;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.stargate.db.datastore.Row;
+import io.stargate.web.docsapi.exception.DocumentAPIRequestException;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.List;
+
+/**
+ * This class is in charge of keeping the data of the Docs API pagination process. It's not only a
+ * stateful component, but a state machine with a initial state. It's created on the REST layer and
+ * passed through to the DocumentService that is in charge of its changes. After being populated its
+ * `documentPageState` can be passed by the REST layer to the client as a JSON string.
+ */
+public class Paginator {
+
+  private final ObjectMapper mapper = new ObjectMapper();
```

Comment:
I am not sure how the object mapper is handled in the project in general, but wouldn't it be better to pass it as the part of the constructor. This way any kind of mapper configuration is lost, I am not sure if this is intended or not?

Reply:
Yes, that's a good point - I didn't notice this before; I will use the same ObjectMapper that the rest of the project uses

Reply:
oof, this is a problem throughout the stargate project, this pattern of creating ObjectMappers is all over SG. I will make a ticket to make this use a single object mapper, it's more organized wrt configuration...

Reply:
Yea, ticket is fine..

Reply:
We just create new ones since so far we haven't had to add custom config for them.

Reply:
This `ObjectMapper` is being removed in https://github.com/stargate/stargate/pull/872